crate_universe/src/utils.rs

```diff
@@ -16,3 +19,94 @@ pub(crate) fn sanitize_module_name(name: &str) -> String {
 pub(crate) fn sanitize_repository_name(name: &str) -> String {
     name.replace('+', "-")
 }
+
+/// We need to sanitize the cargo vendor'ed folder names.  Cargo automatically vendors the crates
+/// based off the semver spec (libfoo-v.1.1.0+MyMeta), thus making bazel labels invalid, since they
+/// dont accept a (+) symbol.  This simpley takes the outputs, and will sanitize the path bufs for
+/// buildifier and for when we write the outputs.
+pub(crate) fn sanitize_vendor_file_names(outputs: &BTreeMap<PathBuf, String>) -> BTreeSet<PathBuf> {
+    outputs
+        .keys()
+        .cloned()
+        .map(|p| {
+            let p_str = sanitize_repository_name(p.to_str().unwrap());
+            PathBuf::from(p_str)
+        })
+        .collect()
+}
+
+/// Vendored crates are generated by cargo itself in `src/metadata.rs` in the
+/// `VendorGenerator::generate()` method.  This means that the semver metadata will
+/// always contain a (+) symbol, which is not compatible with bazel's labels.  
+/// This function will rename the cargo vendor generated file paths to be compatible with bazel
+/// labels by simply replacing the (+) with a (-). If this file is called by any other cli mod,
+/// it just simplies joins the out dir to the path
+pub(crate) fn normalize_cargo_file_paths(
+    outputs: BTreeMap<PathBuf, String>,
+    out_dir: &Path,
+) -> BTreeMap<PathBuf, String> {
+    outputs
```

Comment:
I moved the re mapping of  the outputs from write_outputs into this function, that way the iteration still only happens one time.  So it combines 
```
    let outputs: BTreeMap<PathBuf, String> = outputs
        .into_iter()
        .map(|(path, content)| (out_dir.join(path), content))
        .collect();
```
With the renaming, so we get both the join from  https://github.com/bazelbuild/rules_rust/pull/2579/files#diff-52bdc04b030a02035a19a3c805288b7720191185a18f6b8f16f61a59baa42d3fL787 and renaming of the files in one swoop..  

if nothing needs to be renamed, it simply just does the `out_dir` join, previously in the `write_outputs` function